### PR TITLE
Second Tokenization Add On fix

### PR DESCRIPTION
**Pull request by ujaiswal**
**Description**: Second Tokenization Add On fix

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc [hunk 1]
```diff
@@ -228,8 +231,8 @@ void LogitsProcessorList::Process(const ISequences* sequences,
                                   int step) {
   NextTokenScores<float> input_scores = {next_token_scores, batch_beam_size_, vocab_size_};
   for (size_t i = 0; i < processor_list_.size(); i++) {
-    // Prefix vocab mask is applied to first iteration only.
-    if (step > 1 && processor_list_[i] == prefix_vocab_mask_processor_.get()) {
+    // Prefix vocab mask is applied until max_iteration_for_prefix_vocab_mask is reached.
+    if (processor_list_[i] == prefix_vocab_mask_processor_.get() && step > prefix_vocab_mask_processor_->max_iteration_for_prefix_vocab_mask) {
```
Comment (ViswanathB) [inline onnxruntime/contrib_ops/cpu/transformers/logits_processor.cc:235]: I think it takes care of the prefix masking, but after this flow generated tokens are used to calculate a score (happens in expected character savings) which will also need prefix lengths.

@utjais : prefix lengths changes when you change the prefix, we need to consider that too, would it be an input? Currently prefix length is an input with dimension batch_size, it would need batch_size x max_iteration
